### ISSUE TITLE
[Bug]: [Dropdown] [A11y] Close icon button on the Callout can't access with keyboard.

### ISSUE THREAD

**Issue by vinfol** [microsoft/fluentui#34767]
### Package

react

### Package version

8.123.0

### React version

16.14.0

### Environment

```shell
NA
```

### Current Behavior

When use Dropdown on small screen page, the callout will show on the right side, just like that
![Image](https://github.com/user-attachments/assets/8b0c675e-e429-4399-ba9f-0660f5009d23)
But the close icon button can't access using keyboard

### Expected Behavior

The close icon button should be access using keyboard

### Reproduction

https://codepen.io/vinfol-the-builder/pen/LEVoxJV

### Steps to reproduce

Open https://codepen.io/vinfol-the-builder/pen/LEVoxJV
Then use F12 open devtools
Change to Elements tab, and select the device toolbar to chang Dimensions
![Image](https://github.com/user-attachments/assets/4a171a36-e2de-441e-a7eb-18cbc6115074)

![Image](https://github.com/user-attachments/assets/2ed82450-d6a5-4890-956f-80577a914f05)

![Image](https://github.com/user-attachments/assets/aa49169e-5fdb-430e-b946-eb98b8816ed0)

Click the Dropdown, and then the callout will show on the right, the close icon button can't access using keyboard

### Are you reporting an Accessibility issue?

None

### Suggested severity

High - No workaround

### Are you willing to submit a PR to fix?

yes

### Validations

- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a minimal reproducible example of the bug.

**Comment by Hotell** [comment]
pressing `ESC` works so this is not probably a high severity issues, but AFAIR the close button should be accessible via keyboard so this is a bug

@vinfol happy to accept fix if you have capacity, ty!

**Comment by smhigley** [comment]
The keyboard method to dismiss is `esc`, as @Hotell mentioned. The close button intentionally doesn't have access, as this is the mobile version of a listbox popup, so keyboard focus remains within the listbox.

The close button is primarily there for pointer, voice control, and touchscreen (i.e. no keyboard) screen reader users.
